setup.py

```diff
@@ -956,11 +1024,91 @@ def build_torch_extension_v2(build_ext, options, torch_version):
     build_ext.build_extension(torch_mpi_lib_v2)
 
 
+def build_cmake(build_ext, ext, output_dir, options):
+    # CMake should come with pip requirement 'cmake', but let's verify
+
+    cmake_bin = 'cmake'
+    if not have_cmake:
+        # if the system doesn't have cmake before, we need to check the newly
+        # installed cmake binary is executable.
+        import cmake
+        cmake_bin = os.path.join(cmake.CMAKE_BIN_DIR, 'cmake')
+        if not os.access(cmake_bin, os.X_OK):
+            # if not executable, +x to the binary
+            user_mode = stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR
+            group_mode = stat.S_IRGRP | stat.S_IWGRP | stat.S_IXGRP
+            other_mode = stat.S_IROTH | stat.S_IXOTH
+            os.chmod(cmake_bin, user_mode | group_mode | other_mode)
+
+    # check again if the cmake binary is usable
+    try:
+        subprocess.check_output([cmake_bin, '--version'])
+    except OSError as e:
+        raise RuntimeError('Check CMake executable failed: {}'.format(str(e)))
+
+    # Statically linked archive files go into the provided output directory
+    extdir = os.path.abspath(
+        os.path.dirname(build_ext.get_ext_fullpath(ext.name)))
+    config = 'Debug' if build_ext.debug else 'Release'
+    cmake_args = [
+        '-DUSE_MPI=ON',
+        '-DCMAKE_BUILD_TYPE=' + config,
+        '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY_{}={}'.format(config.upper(), extdir),
+        '-DCMAKE_ARCHIVE_OUTPUT_DIRECTORY_{}={}'.format(config.upper(),
+                                                        output_dir),
+    ]
+    cmake_build_args = [
+        '--config', config,
+        '--', '-j4',
+    ]
+
+    # Keep temp build files within a unique subdirectory
+    build_temp = os.path.abspath(os.path.join(build_ext.build_temp, ext.name))
+    if not os.path.exists(build_temp):
+        os.makedirs(build_temp)
+
+    # Config and build the extension
+    try:
+        subprocess.check_call([cmake_bin, ext.cmake_lists_dir] + cmake_args,
+                              cwd=build_temp)
+        subprocess.check_call([cmake_bin, '--build', '.'] + cmake_build_args,
+                              cwd=build_temp)
+    except OSError as e:
+        raise RuntimeError('CMake failed: {}'.format(str(e)))
+    # Add the library so other extensions will link against it during compilation
```

Comment:
Add a blank line before comment for clarity

Reply:
Fixed.